openlibrary/plugins/openlibrary/js/ol.js

```diff
@@ -75,48 +79,263 @@ export function closePop(){
     });
 }
 
-export default function init(){
-    var $searchResults = $('header#header-bar .search-component ul.search-results');
-    var $searchInput = $('header#header-bar .search-component .search-bar-input input[type="text"]');
-    var cover_url = function(id) {
-        return `//covers.openlibrary.org/b/id/${id}-S.jpg`
-    };
-    // stores the state of the search result for resizing window
-    var instantSearchResultState = false;
-    var searchModes, searchModeDefault, defaultFacet, searchFacets, composeSearchUrl, marshalBookSearchQuery, renderInstantSearchResults, setFacet, setMode, setSearchMode, options, q, parts, enteredSearchMinimized, searchExpansionActivated, toggleSearchbar, renderInstantSearchResult, val, facet_value;
-
-    // searches should be cancelled if you click anywhere in the page
-    $('body').on('click', function () {
-        instantSearchResultState = false;
-        $searchResults.empty();
-    });
-    // but clicking search input should not empty search results.
-    $searchInput.on('click', false);
+const SEARCH_MODES = ['everything', 'ebooks', 'printdisabled'];
+const SEARCH_MODE_DEFAULT = 'ebooks';
+const SEARCH_FACETS = {
+    'title': 'books',
+    'author': 'authors',
+    'lists': 'lists',
+    'subject': 'subjects',
+    'all': 'all',
+    'advanced': 'advancedsearch',
+    'text': 'inside'
+};
+const DEFAULT_SEARCH_FACET = 'all';
+const RENDER_INSTANT_SEARCH_RESULT = {
```

Comment:
why capitals letters? Is this becoming a code convention? I don't know of any repos that do this, so I'd say this is an unusual choice that needs justification.

From my perspective, JavaScript doesn't allow you to reassign to a constant - it throws an error, so there should be no concern of misusing one.
FWIW I don't like capitalising these as if you need to ever change a const to a let, you have to change every variable usage and that ruins git blame and causes unnecessary hassle.

Reply:
It's pretty common: 
- Google: https://google.github.io/styleguide/jsguide.html#naming-constant-names
- AirBnB: https://github.com/airbnb/javascript#naming--uppercase

Note `const` just denotes a runtime constant; using uppercase denotes a build-time constant. The google docs describe it well. It's useful when you see a it used in the code, because you know it doesn't depend on anything. You could programatically find-and-replace all occurences with the value, and everything would work. (It's basically a macro).

Personally, I've used this convention at a few places I've worked at and find it valuable.

ex:
![image](https://user-images.githubusercontent.com/6251786/63219876-9d756280-c149-11e9-9a4f-f978d75a863d.png)

Looking at this code, you know that `DEFAULT_FACET` is not only `const`, but that it's defined at build time; it doesn't depend on the users preferences, or on a url parameter; it's some global fixed value.

Reply:
I think it makes sense  for exported variables (per Airbnb) suggestion but not for all const variables (the google way)

e.g. 
```
export const DEFAULT_FACET
```

In the airbnb link you provide it says:
"what about all const variables? - This is unnecessary, so uppercasing should not be used for constants within a file. It should be used for exported constants however."

I agree it's unnecessary.

Forgetting default is a keyword for a minute in the example above consider the readability in:
```
const default = DEFAULT_FACET;
this.facet = new PersistentValue( 'facet', { default } );
```
I would like us to retain the flexibility of using lowercase const during the refactor portion of this project in case we ever need to change them to lets.

Reply:
As a compromise can we move these to a separate file and export them from it so they are reusable elsewhere?

e.g. `js/search/config.js`

Reply:
I'm not crazy about that; that's coupling a bunch of entirely unrelated values which are unlikely to be reused outside of the context of their respective files. I've had experience with this convention and I find it provides value; it's also used in the python. I'm not going to work on developing a style guide rn, but this is something we can trivially (and robotically) change if we want to later on, so going for WONTFIX.